File: java/parser/src/main/java/gov/lbl/parser/parser/Component_listVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Component_declaration;
import gov.lbl.parser.domain.Component_list;

import static java.util.stream.Collectors.toList;

public class Component_listVisitor extends modelicaBaseVisitor<Component_list> {
    @Override
    public Component_list visitComponent_list(modelicaParser.Component_listContext ctx) {
        Component_declarationVisitor component_declarationVisitor = new Component_declarationVisitor();
        List<Component_declaration> component_declaration_list = ctx.component_declaration()
                                                                    .stream()
                                                                    .map(component_declaration -> component_declaration.accept(component_declarationVisitor))
                                                                    .collect(toList());
        return new Component_list(component_declaration_list);
    }
}

File: jsParser/parser/Component_listVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Component_list = require('../domain/Component_list')

const Component_declarationVisitor = require('./Component_declarationVisitor')

class Component_listVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitComponent_list (ctx) {
    const component_declaration_list = []

    if (ctx.component_declaration()) {
      const component_declarationVisitor = new Component_declarationVisitor.Component_declarationVisitor()
      ctx.component_declaration().forEach(component_dec => {
        component_declaration_list.push(component_declarationVisitor.visitComponent_declaration(component_dec))
      })
    }

    return new Component_list.Component_list(component_declaration_list)
  }
};

Component_listVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitComponent_list = this.visitComponent_list
exports.Component_listVisitor = Component_listVisitor
